src/OpenTelemetry/Trace/Span.cs

```diff
@@ -174,92 +159,79 @@ public Status Status
         /// <inheritdoc />
         public void UpdateName(string name)
         {
+            if (this.hasEnded)
+            {
+                OpenTelemetrySdkEventSource.Log.UnexpectedCallOnEndedSpan("UpdateName");
+                return;
+            }
+
             this.Name = name ?? throw new ArgumentNullException(nameof(name));
         }
 
         /// <inheritdoc/>
         public void SetAttribute(KeyValuePair<string, object> keyValuePair)
         {
-            if (keyValuePair.Key == null || keyValuePair.Value == null)
+            if (keyValuePair.Key == null)
```

Comment:
A general comment about all of the parameter validations we are putting in.

Seems to violate the basic error handling principles outlined here:
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/error-handling.md

No?

Reply:
Cool, I didn't know that we have error handling principles. You are right.  This is much wider that Span only, I created a new issue to track it https://github.com/open-telemetry/opentelemetry-dotnet/issues/295.

Reply:
thanks!